src/app/login/login.component.ts

```diff
@@ -0,0 +1,37 @@
+import { Component, OnInit } from '@angular/core';
+import { FormBuilder, FormGroup, Validators } from '@angular/forms';
+import { Router } from '@angular/router';
+import { User } from '../module/user';
+
+@Component({
+  selector: 'app-login',
+  templateUrl: './login.component.html',
+  styleUrls: ['./login.component.css']
+})
+export class LoginComponent implements OnInit {
+
+  logForm: FormGroup = new FormGroup({});
+  submitted : boolean = false;
+
+  constructor(private formBuilder: FormBuilder, private router: Router) { }
+
+  ngOnInit(): void {
+    this.logForm = this.formBuilder.group({
+      username: ['', [Validators.required, Validators.minLength(10), Validators.maxLength(10)]],
+      password: ['', [Validators.required, Validators.minLength(8)]],
+    });
+  }
+
+  onSubmit(): void {
+    this.submitted = true;
+    
+    if (this.logForm.valid) {
+      let user: User = this.logForm.value;
+      console.log(user);
+      this.router.navigate(['/home']);
```

Comment:
I recommend reading paths from a file. you can create a file for app's constants that has routes, constants, etc